src/components/TodoItem.tsx

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable no-param-reassign */
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  changeTodoStatus: (completed: boolean, todoId:number) => void;
+  deleteTodo: (todoId: number) => void;
+  editTitle: (todoTitle: string, todoId: number) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  changeTodoStatus,
+  deleteTodo,
+  editTitle,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  return (
+    <li
+      className={classNames({ completed: todo.completed }, { editing })}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          checked={todo.completed}
+          onChange={() => {
+            const completed = !todo.completed;
+
+            changeTodoStatus(completed, todo.id);
+          }}
+        />
+        <label
+          role="presentation"
+          onKeyDown={() => {}}
+          onDoubleClick={() => {
+            setEditing(true);
+          }}
```

Comment:
`onKeyDown` should have the same handler as `onDoubleClick`, read more - https://dev.to/receter/easy-accessible-click-handlers-4jkb